R/word.R

```diff
@@ -1,24 +1,38 @@
-#' @rdname html_document2
-#' @export
-word_document2 = function(fig_caption = TRUE, md_extensions = NULL, pandoc_args = NULL, ...) {
-  from = rmarkdown::from_rmarkdown(fig_caption, md_extensions)
+office_document2_base = function(base_format) {
```

Comment:
This is good, but I slightly prefer this style:

```r
office_document2_base = function(
  fig_caption = TRUE, md_extensions = NULL, pandoc_args = NULL, ..., base_format
) {

}
```

It will be more consistent with other functions like
https://github.com/rstudio/bookdown/blob/9516a435773fc55d822d10012101405af54f6274/R/html.R#L123-L125

What's more, we can export this function and allow users to customize all arguments of `base_format()`.

Actually, let's rename it to `markdown_document2()` (because it is not only useful to Office documents) and export it. You will also need to document it. It may take you some time to explain what this function really does.

Reply:
Got it. So updated `word_document2` will be

``` r
word_document2 = function(
  ..., fig_caption = TRUE, md_extensions = NULL, pandoc_args = NULL
) {
  markdown_document2(
    fig_caption = fig_caption, md_extensions = md_extensions, pandoc_args = pandoc_args,
    ..., base_format = rmarkdown::word_document
  )
}
```

or more simply like `html_book`,

``` r
word_document2 = function(...) markdown_document2(..., base_format = rmarkdown::word_document)
```

Which do you prefer?

Do I document `markdown_document2` in `@rdname html_document2` or in an individual file?

Reply:
1. The latter.

2. You can do `@rdname html_document2`.

Reply:
Thanks and I did it :)